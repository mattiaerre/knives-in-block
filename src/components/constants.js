export const BLACK = 'black';
export const DOWN = 'down';
export const EMPTY = 'empty';
export const UP = 'up';
