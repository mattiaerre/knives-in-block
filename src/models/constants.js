export const DOWN = 'down';
export const EMPTY = 'empty';
export const FULL = 'full';
export const UP = 'up';
