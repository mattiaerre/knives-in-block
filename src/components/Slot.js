import { BLACK, EMPTY } from './constants';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './Slot.module.css';

export default function Slot({ id, knife, onClick }) {
  return (
    <div
      className={classNames(styles[knife ? knife.state : EMPTY], styles.slot)}
      data-testid={id}
      id={id}
      onClick={() => {
        onClick({ id: id, knife: knife });
      }}
      style={{ backgroundColor: knife ? knife.color : BLACK }}
    ></div>
  );
}

Slot.propTypes = {
  id: PropTypes.string.isRequired,
  knife: PropTypes.object,
  onClick: PropTypes.func.isRequired
};
