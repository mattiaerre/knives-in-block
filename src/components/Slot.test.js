import { DOWN } from './constants';
import Slot from './Slot';
import { render } from '@testing-library/react';

it('renders w/ knife', () => {
  const { asFragment } = render(
    <Slot
      id="s1"
      knife={{ color: 'blue', id: 'k1', state: DOWN }}
      onClick={() => {}}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it('renders w/o knife', () => {
  const { asFragment } = render(<Slot id="s2" onClick={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
