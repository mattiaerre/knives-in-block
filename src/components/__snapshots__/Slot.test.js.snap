// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders w/ knife 1`] = `
<DocumentFragment>
  <div
    class="undefined slot"
    data-testid="s1"
    id="s1"
    style="background-color: blue;"
  />
</DocumentFragment>
`;

exports[`renders w/o knife 1`] = `
<DocumentFragment>
  <div
    class="empty slot"
    data-testid="s2"
    id="s2"
    style="background-color: black;"
  />
</DocumentFragment>
`;
